FileShareService.DesktopClient.Core/TokenCacheHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.Identity.Client;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace UKHO.FileShareService.DesktopClient.Core
+{
+    static class TokenCacheHelper
+    {
+        static TokenCacheHelper()
+        {
+            try
+            {
+                // For packaged desktop apps (MSIX packages, also called desktop bridge) the executing assembly folder is read-only. 
+                // In that case we need to use Windows.Storage.ApplicationData.Current.LocalCacheFolder.Path + "\msalcache.bin" 
+                // which is a per-app read/write folder for packaged apps.
+                // See https://docs.microsoft.com/windows/msix/desktop/desktop-to-uwp-behind-the-scenes
+                //  CacheFilePath = Path.Combine(Windows.Storage.ApplicationData.Current.LocalCacheFolder.Path, "msalcache.bin");
+                CacheFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "msalcache.bin");
+                // CacheFilePath = Path.Combine(@"D:\FSSTooling\TestPath", "msalcache.bin");
+            }
+            catch (System.InvalidOperationException)
```

Comment:
Path.Combine won't throw an exception, so only `Environment.GetFolderPath()` can throw.  It will only throw `ArgumentException` if folder is not a member of Environment.SpecialFolder or `PlatformNotSupportedException` if The current platform does not support the specified folder, neither of which is an `InvalidOperationException`.

This code look cut'n'paste.  Could you please work though it, with the appropriate modifications and ensure it is correct please.

Reply:
Resolved